utils/logalloc.cc

```diff
@@ -830,7 +830,10 @@ class segment_pool {
     segment_pool();
     void prime(size_t available_memory, size_t min_free_memory);
     segment* new_segment(region::impl* r);
-    segment_descriptor& descriptor(segment*) noexcept;
+    segment_descriptor& descriptor(segment* seg) noexcept {
+        uintptr_t index = idx_from_segment(seg);
+        return _segments[index];
+    }
```

Comment:
it's just a step towards the next patch that adds a const overload of descriptor().
I thought it would help make reviewing of the patches a little easier.